Tools/PIDML/pidMLEffAndPurProducer.cxx

```diff
@@ -0,0 +1,212 @@
+  Configurable<double> cfgNSigmaCut{"n-sigma-cut", 3.0f, "TPC and TOF PID nSigma cut"};
+  Configurable<double> cfgTofPtCut{"tof-pt-cut", 0.5f, "From what pT TOF is used"};
+  Configurable<double> cfgCertainty{"certainty", 0.5, "Min certainty of the model to accept given mcPart to be of given kind"};
+
+  Configurable<std::string> cfgPathCCDB{"ccdb-path", "Users/m/mkabus/PIDML", "base path to the CCDB directory with ONNX models"};
+  Configurable<std::string> cfgCCDBURL{"ccdb-url", "http://alice-ccdb.cern.ch", "URL of the CCDB repository"};
+  Configurable<bool> cfgUseCCDB{"useCCDB", true, "Whether to autofetch ML model from CCDB. If false, local file will be used."};
+  Configurable<std::string> cfgPathLocal{"local-path", "/home/mkabus/PIDML", "base path to the local directory with ONNX models"};
+
+  Configurable<bool> cfgUseFixedTimestamp{"use-fixed-timestamp", false, "Whether to use fixed timestamp from configurable instead of timestamp calculated from the data"};
+  Configurable<uint64_t> cfgTimestamp{"timestamp", 1524176895000, "Hardcoded timestamp for tests"};
+
+  static constexpr float kTOFMissingSignal = -999.0f;
+  static constexpr float kTRDMissingSignal = 0.0f;
+  static constexpr float kEtaCut = 0.8f;
+  static constexpr float kEpsilon = 1e-10f;
+
+  o2::ccdb::CcdbApi ccdbApi;
+  int currentRunNumber = -1;
+
+  Filter trackFilter = requireGlobalTrackInFilter();
+
+  using BigTracks = soa::Filtered<soa::Join<aod::FullTracks, aod::TracksDCA, aod::pidTOFbeta, aod::TrackSelection, aod::TOFSignal, aod::McTrackLabels,
+                                            aod::pidTPCFullPi, aod::pidTPCFullKa, aod::pidTPCFullPr, aod::pidTPCFullEl, aod::pidTPCFullMu,
+                                            aod::pidTOFFullPi, aod::pidTOFFullKa, aod::pidTOFFullPr, aod::pidTOFFullEl, aod::pidTOFFullMu>>;
+
+  typedef struct nSigma_t {
+    double tpc, tof;
+  } nSigma_t;
+
+  nSigma_t GetNSigma(const BigTracks::iterator& track)
+  {
+    nSigma_t nSigma;
+
+    switch (TMath::Abs(cfgPid)) {
+      case 11: // electron
+        nSigma.tof = track.tofNSigmaEl();
+        nSigma.tpc = track.tpcNSigmaEl();
+        break;
+      case 13: // muon
+        nSigma.tof = track.tofNSigmaMu();
+        nSigma.tpc = track.tpcNSigmaMu();
+        break;
+      case 211: // pion
+        nSigma.tof = track.tofNSigmaPi();
+        nSigma.tpc = track.tpcNSigmaPi();
+        break;
+      case 321: // kaon
+        nSigma.tof = track.tofNSigmaKa();
+        nSigma.tpc = track.tpcNSigmaKa();
+        break;
+      case 2212: // proton
+        nSigma.tof = track.tofNSigmaPr();
+        nSigma.tpc = track.tpcNSigmaPr();
+        break;
+    }
+
+    return nSigma;
+  }
+
+  bool IsNSigmaAccept(const BigTracks::iterator& track, nSigma_t& nSigma)
+  {
+    int sign = cfgPid > 0 ? 1 : -1;
+    if (track.sign() != sign)
+      return false;
+
+    if (track.pt() <= cfgTofPtCut || (track.tofSignal() - kTOFMissingSignal) < kEpsilon) {
```

Comment:
(not important) keep one convention - in batch there is exclusive pt cut, just to be sure both scripts would give same results